research/object_detection/core/box_list_ops.py

```diff
@@ -303,6 +303,49 @@ def iou(boxlist1, boxlist2, scope=None):
         tf.equal(intersections, 0.0),
         tf.zeros_like(intersections), tf.truediv(intersections, unions))
 
+def l1(boxlist1, boxlist2, scope=None):
+  """Computes l1 loss (pairwise) between two boxlists.
+
+  Args:
+    boxlist1: BoxList holding N boxes
+    boxlist2: BoxList holding M boxes
+    scope: name scope.
+
+  Returns:
+    a tensor with shape [N, M] representing the pairwise L1 loss.
+  """
+  with tf.name_scope(scope, 'PairwiseL1'):
+    ycenter1, xcenter1, h1, w1 = boxlist1.get_center_coordinates_and_sizes()
+    ycenter2, xcenter2, h2, w2 = boxlist2.get_center_coordinates_and_sizes()
+    ycenters = tf.abs(tf.expand_dims(ycenter2, axis=0) - \
+        tf.expand_dims(tf.transpose(ycenter1), axis=1))
+    xcenters = tf.abs(tf.expand_dims(xcenter2, axis=0) - \
+        tf.expand_dims(tf.transpose(xcenter1), axis=1))
+    heights = tf.abs(tf.expand_dims(h2, axis=0) - \
+        tf.expand_dims(tf.transpose(h1), axis=1))
+    widths = tf.abs(tf.expand_dims(w2, axis=0) - \
+        tf.expand_dims(tf.transpose(w1), axis=1))
+    return ycenters + xcenters + heights + widths
+
+def giou_loss(boxlist1, boxlist2, scope=None):
```

Comment:
It is already available at https://www.tensorflow.org/addons/api_docs/python/tfa/losses/GIoULoss
If you have any extension please open a PR in TFA. /cc @tensorflow/sig-io-maintainers

Reply:
@tombstone told me to implement this separately so we wouldn't need that dependency.

Reply:
Ok thanks. 
@tombstone Our check it is just part of the approved RFC at https://github.com/tensorflow/community/blame/master/rfcs/20190802-model-garden-redesign.md#L60-L61

Reply:
Another GIOU Is landing from https://github.com/tensorflow/models/pull/9083